### ISSUE TITLE
Can not show correct 404 page in some cases

### ISSUE THREAD

**Issue by tjeeay** [nuxt/nuxt.js#7894]
### Versions

- nuxt: 
- node: 

### Reproduction

when i access the page url that does't exists in my website, nuxt.js does not display the correct 404 page as expected.

### Steps to reproduce

directly access these url in browser:
- [http://localhost:3000/test.txt](http://localhost:3000/test.txt)
- [http://localhost:3000/test.js](http://localhost:3000/test.js)
- [http://localhost:3000/test.css](http://localhost:3000/test.css)

### What is Expected?
seems in most cases, the 404 page will be displayed as expected.
![image](https://user-images.githubusercontent.com/19549395/89972247-ec3bd680-dc8f-11ea-8a2f-e2baa42e024d.png)

### What is actually happening?
but, if the url end with `.txt`, '.js' or '.css', the result is not what i want.
![image](https://user-images.githubusercontent.com/19549395/89972092-89e2d600-dc8f-11ea-8838-c2d43821d1dc.png)
![image](https://user-images.githubusercontent.com/19549395/89972139-9ebf6980-dc8f-11ea-8eeb-040d073bde0f.png)
![image](https://user-images.githubusercontent.com/19549395/89972159-a979fe80-dc8f-11ea-848a-daaf912c2bad.png)

**Comment by ngblaylock** [comment]
This may or may not be related to your specific issue...

I updated to Nuxt 2.14.1, and when I try to get to the error page when it is hosted somewhere, it just had a spinner and wouldn't load the error page. It worked just fine in development and after building and running the start command. I rolled back to 2.13.3 and ran the build and export, then pushed it live and it worked just fine.

I see quite a few issues with 2.14 that are somewhat related to generating error pages correctly. Hopefully we get a fix soon. I am not sure exactly what the issue is.

I tested the hosting on GitHub Pages, and Firebase Hosting.
